feature_engine/sanity_check/similar_columns.py

```diff
@@ -0,0 +1,125 @@
+from typing import Any
+
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+
+from feature_engine.validation import _return_tags
+
+from feature_engine.dataframe_checks import (
+    _check_contains_inf,
+    _check_contains_na,
+    _is_dataframe,
+)
+from feature_engine.variable_manipulation import _find_all_variables
+
+
+class SimilarColumns(BaseEstimator, TransformerMixin):
+    """Ensure that similar columns are in test and train dataset.
+
+    Parameters
+    ----
+    None
+
+    """
+
+    def __init__(
+        self,
+        fill_value: Any = np.nan,
```

Comment:
fill_value should take np.nan, integers or strings. Not Any. Could you please fix?